backend/src/__tests__/events/debug/addDebugEmitAfterEachEmit.test.ts

```diff
@@ -0,0 +1,98 @@
+
+import { addDebugEmitAfterEachEmit } from "../../../debug/addDebugEmitAfterEachEmit"
+
+import { createServer } from 'node:http';
+import { AddressInfo } from 'net';
+import { io as ioc, Socket as ClientSocket } from 'socket.io-client';
+import { Server, Socket as ServerSocket } from 'socket.io';
+import { lobbys, players } from '../../../global';
+
+const httpServer = createServer();
+let io: Server;
+let clientSocket: ClientSocket;
+let serverSocket: ServerSocket;
+
+function instancianteSockets(onDone: () => void, useAddDebug: boolean) {
+  io = new Server(httpServer);
+  if (useAddDebug) {
+    addDebugEmitAfterEachEmit(io)  // Essa função adiciona um middleware que emite um evento 'debug' após cada evento emitido. Por isso precisa ser chamada antes de instanciar os sockets.
+  }
+  httpServer.listen(() => {
+    const port = (httpServer.address() as AddressInfo).port;
+    clientSocket = ioc(`http://localhost:${port}`);
+    io.on('connection', (socket) => {
+      serverSocket = socket;
+      onDone()
+    });
+    clientSocket.on('connect', () => {});
+  });
+}
+
+afterEach(() => {
+  for (const lobby in lobbys) {
+    delete lobbys[lobby];
+  }
+  for (const player in players) {
+    delete players[player];
+  }
+  clientSocket.removeAllListeners();
+  clientSocket.removeAllListeners('player-join');
+
+  httpServer.close();
+  io.close();
+  clientSocket.disconnect();
+});
+
+describe('addDebugEmitAfterEachEmit', () => {
+
+  test('Deve emitir evento debug após cada evento emitido', (done) => {
+    instancianteSockets(() => {
+      players[serverSocket.id] = {
+        socket: serverSocket
+      }
+
+      serverSocket.emit('test')
+      clientSocket.on('debug', (lobby: any) => {
```

Comment:
Uma boa pratica é não definir variáveis/parâmetros que não serão utilizadas, entretanto, várias vezes é interessante ou necessário informar quais são os parâmetros da função.

Para isso, um padrão adotado é utilizar um _ (_underline_) no inicio do parâmetro para informar que ele não é importante e não será utilizado nesse contexto.

(Isso irá remover alguns problemas de lint que não podem ser resolvidos automaticamente)

```suggestion
      clientSocket.on('debug', (_lobby: Lobby) => {
```

Outra opção, é só não definir esse parâmetro, mas perdemos um pouco de informação e padronização.

```suggestion
      clientSocket.on('debug', () => {
```